src/util/locking/GreedyReadWriteLocker.ts

```diff
@@ -0,0 +1,140 @@
+import type { ResourceIdentifier } from '../../ldp/representation/ResourceIdentifier';
+import type { MapStorage } from '../../storage/mapstorage/MapStorage';
+import { ForbiddenHttpError } from '../errors/ForbiddenHttpError';
+import type { ReadWriteLocker } from './ReadWriteLocker';
+import type { ResourceLocker } from './ResourceLocker';
+
+export interface GreedyReadWriteSuffixes {
+  count: string;
+  read: string;
+  write: string;
+}
+
+/**
+ * A {@link ReadWriteLocker} that allows for multiple simultaneous read operations.
+ * Write operations will be blocked as long as read operations are not finished.
+ * New read operations are allowed while this is going on, which will cause write operations to wait longer.
+ */
+export class GreedyReadWriteLocker implements ReadWriteLocker {
```

Comment:
Maybe document algorithm?

Reply:
This was my (too naive I guess 😄 ) at documenting the algorithm. I can add a link to the wikipedia algorithm? Doesn't seem nice to just copy/paste that.

Reply:
Makes sense.